MicroProfile Health quickstart is failing

It looks related to the latest changes that were merged (https://github.com/quarkusio/quarkus/pull/4987).

@loicmathieu @jmartisk could you have a look?

It's executing:
```
+ cd microprofile-fault-tolerance-quickstart
+ mvn clean install -Pnative
```
but fails right in the Surefire tests so it's not a native specific issue.

```
2019-11-25 09:53:52,574 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (vert.x-worker-thread-2) HTTP Request to /coffee/2/recommendations failed, error id: 753d6bb7-761f-4121-8e6b-89d4799c421d-2: org.jboss.resteasy.spi.UnhandledException: java.lang.NoClassDefFoundError: io/quarkus/smallrye/faulttolerance/deployment/NoopMetricRegistry
	at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:106)
	at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:372)
	at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:209)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:496)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:252)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:153)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:363)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:156)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:238)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:108)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.lambda$handle$0(VertxRequestHandler.java:73)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:316)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NoClassDefFoundError: io/quarkus/smallrye/faulttolerance/deployment/NoopMetricRegistry
	at io.quarkus.smallrye.faulttolerance.deployment.NoopMetricRegistry_Bean.create(NoopMetricRegistry_Bean.zig:61)
	at io.quarkus.smallrye.faulttolerance.deployment.NoopMetricRegistry_Bean.create(NoopMetricRegistry_Bean.zig:23)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:80)
	at io.quarkus.arc.impl.ComputingCache$CacheFunction.lambda$apply$0(ComputingCache.java:99)
	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
	at io.quarkus.arc.impl.ComputingCache.getValue(ComputingCache.java:41)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:25)
	at io.quarkus.smallrye.faulttolerance.deployment.NoopMetricRegistry_Bean.get(NoopMetricRegistry_Bean.zig:214)
	at io.quarkus.smallrye.faulttolerance.deployment.NoopMetricRegistry_Bean.get(NoopMetricRegistry_Bean.zig:170)
	at io.smallrye.faulttolerance.metrics.MetricsCollectorFactory_Bean.create(MetricsCollectorFactory_Bean.zig:356)
	at io.smallrye.faulttolerance.metrics.MetricsCollectorFactory_Bean.create(MetricsCollectorFactory_Bean.zig:32)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:80)
	at io.quarkus.arc.impl.ComputingCache$CacheFunction.lambda$apply$0(ComputingCache.java:99)
	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
	at io.quarkus.arc.impl.ComputingCache.getValue(ComputingCache.java:41)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:25)
	at io.smallrye.faulttolerance.metrics.MetricsCollectorFactory_ClientProxy.arc$delegate(MetricsCollectorFactory_ClientProxy.zig:110)
	at io.smallrye.faulttolerance.metrics.MetricsCollectorFactory_ClientProxy.createCollector(MetricsCollectorFactory_ClientProxy.zig:251)
	at io.smallrye.faulttolerance.HystrixCommandInterceptor.executeCommand(HystrixCommandInterceptor.java:265)
	at io.smallrye.faulttolerance.HystrixCommandInterceptor.interceptCommand(HystrixCommandInterceptor.java:257)
	at io.smallrye.faulttolerance.HystrixCommandInterceptor_Bean.intercept(HystrixCommandInterceptor_Bean.zig:83)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
	at org.acme.faulttolerance.CoffeeResource_Subclass.recommendations(CoffeeResource_Subclass.zig:445)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:151)
	at org.jboss.resteasy.core.MethodInjectorImpl.lambda$invoke$3(MethodInjectorImpl.java:122)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:602)
	at java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:614)
	at java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:1983)
	at java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:110)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:122)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:594)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:468)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:421)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:363)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:423)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:391)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invoke$1(ResourceMethodInvoker.java:365)
	at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:981)
	at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2124)
	at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:110)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:365)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:477)
	... 13 more
Caused by: java.lang.ClassNotFoundException: io.quarkus.smallrye.faulttolerance.deployment.NoopMetricRegistry
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 61 more
```

--- a/extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/SmallRyeFaultToleranceProcessor.java
+++ b/extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/SmallRyeFaultToleranceProcessor.java
@@ -8,6 +8,7 @@ import java.util.Set;
 
 import javax.annotation.Priority;
 import javax.inject.Inject;
+import javax.inject.Singleton;
 
 import org.eclipse.microprofile.faulttolerance.Asynchronous;
 import org.eclipse.microprofile.faulttolerance.Bulkhead;
@@ -49,6 +50,7 @@ import io.quarkus.deployment.builditem.SystemPropertyBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.NativeImageSystemPropertyBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 import io.quarkus.deployment.logging.LogCleanupFilterBuildItem;
+import io.quarkus.smallrye.faulttolerance.runtime.NoopMetricRegistry;
 import io.quarkus.smallrye.faulttolerance.runtime.QuarkusFallbackHandlerProvider;
 import io.quarkus.smallrye.faulttolerance.runtime.QuarkusFaultToleranceOperationProvider;
 import io.quarkus.smallrye.faulttolerance.runtime.SmallryeFaultToleranceRecorder;
@@ -150,7 +152,8 @@ public class SmallRyeFaultToleranceProcessor {
 
         if (!capabilities.isCapabilityPresent(Capabilities.METRICS)) {
             //disable fault tolerance metrics with the MP sys props and provides a No-op metric registry.
-            additionalBean.produce(new AdditionalBeanBuildItem(NoopMetricRegistry.class));
+            additionalBean.produce(AdditionalBeanBuildItem.builder().addBeanClass(NoopMetricRegistry.class).setRemovable()
+                    .setDefaultScope(DotName.createSimple(Singleton.class.getName())).build());
             systemProperty.produce(new SystemPropertyBuildItem("MP_Fault_Tolerance_Metrics_Enabled", "false"));
         }
     }

--- a/extensions/smallrye-fault-tolerance/runtime/src/main/java/io/quarkus/smallrye/faulttolerance/runtime/NoopMetricRegistry.java
+++ b/extensions/smallrye-fault-tolerance/runtime/src/main/java/io/quarkus/smallrye/faulttolerance/runtime/NoopMetricRegistry.java
@@ -1,14 +1,11 @@
-package io.quarkus.smallrye.faulttolerance.deployment;
+package io.quarkus.smallrye.faulttolerance.runtime;
 
 import java.util.Map;
 import java.util.SortedMap;
 import java.util.SortedSet;
 
-import javax.inject.Singleton;
-
 import org.eclipse.microprofile.metrics.*;
 
-@Singleton
 public class NoopMetricRegistry extends MetricRegistry {
 
     @Override

